lib/execute/src/memory.rs

```diff
@@ -53,15 +53,29 @@ impl LinearMemory {
             Some(new_pages) => new_pages,
             None => return None,
         };
+        if let Some(val) = self.maximum {
+            if new_pages > val {
+                return None;
+            }
+        }
```

Comment:
If `maximum` is unset then it's possible to allocate more than 4GB?

Reply:
That's an interesting question. I think it can address up to 8, with the index plus the offset, and I don't see anything in the spec which limits how much can be allocated.

Reply:
Oh, that's interesting. I always thought that there is a limit, but I was unable to find anything regarding this in the spec nor in spec interpreter.

So: the wasm spec doesn't limit the amount of allocatable (however, embedder can), but there is an inherent limitation on how much memory we can address (and yes, you're right again, it's 8 not 4GBs).

Reply:
Interesting tidbit about this I stumbled upon with my toy interpreter: The official testsuite has a test that claims that allocating more than 4GiB should fail validation, despite the spec enforcing no such limit there. See here: https://github.com/WebAssembly/spec/blob/master/test/core/memory.wast#L52

Reply:
It [turned out](https://github.com/WebAssembly/spec/issues/861#issuecomment-415002493) that you can't actually allocate more than 4GB for an linear memory instance!

Reply:
Cool, thanks for reporting that issue and getting the answer!